lib/utils/index.js

```diff
@@ -11,7 +11,27 @@ function compose() {
   return funcs.reverse().reduce((a, b) => (...arg) => b(a(...arg)));
 }
 
+function getPropertyNames (obj) {
+  if (obj == null) return [];
+  const propertyNames = [];
+  // avoid to deep clone obj
+  let muteObj = Object.getPrototypeOf(obj);
+  do {
+    propertyNames.push.apply(propertyNames, Object.getOwnPropertyNames(muteObj));
+    muteObj = Object.getPrototypeOf(muteObj);
+  } while (muteObj);
```

Comment:
We can stop early if `muteObj.constructor.name === 'Bone'` or `muteObj.constructor.attributes` exists